symphony/lib/toolkit/class.datasource.php

```diff
@@ -536,6 +536,38 @@ public static function removeEscapedCommas($string)
         return preg_replace('/(?<!\\\\)\\\\,/', ',', $string);
     }
 
+    private static $digitReplacements = array(
+        '1' => 'one',
+        '2' => 'two',
+        '3' => 'three',
+        '4' => 'four',
+        '5' => 'five',
+        '6' => 'six',
+        '7' => 'seven',
+        '8' => 'eight',
+        '9' => 'nine',
+        '0' => 'zero',
```

Comment:
Although PHP (still) allows this, we shouldn't have a trailing comma on the last array element. It's just nothing we should encorage, IMHO.

Reply:
@michael-e 
> lthough PHP (still) allows this

Why "still"? More is coming: https://wiki.php.net/rfc/trailing-comma-function-calls 😉

Reply:
Oh. :-(

So what is "good practice" then? I haven't found anything in the PSR-2 coding style guide.

Reply:
> I haven't found anything in the PSR-2 coding style guide.

Right, it's not included.

> So what is "good practice" then?

It's included in the coding standards of many frameworks and IDE's also support automatic adding.

Reply:
Sorry if my question was unclear. I'd like to know which style — trailing comma or not — is the preferred version in PHP and/or Symphony. I know that both styles work, but we should have a preferred one.

Reply:
> The comma after the last array element is optional and can be omitted. This is usually done for single-line arrays, i.e. array(1, 2) is preferred over array(1, 2, ). For multi-line arrays on the other hand the trailing comma is commonly used, as it allows easier addition of new elements at the end.

http://de2.php.net/manual/en/language.types.array.php

> Add a comma after each array item in a multi-line array, even after the last one;

https://symfony.com/doc/current/contributing/code/standards.html#structure

> ZendCodingStandard.Arrays.TrailingArrayComma (requires comma after last element in multiline arrays)

https://github.com/zendframework/zend-coding-standard/pull/1

And I prefer the trailing comma! 😄

Reply:
> It's just nothing we should encorage, IMHO.

I like it. It makes diff of adding/removing element a lot nicer.

We should also encourage early returns.